### DISCUSSION TITLE
ECONNRESET when deleting lots of records

### DISCUSSION THREAD

**Discussion by 0x1AE7F** [pocketbase/pocketbase#2653] [Q&A] [answered]
Im trying to clear a table which has about 1k records in it, containing 5 text fields and one JSON field. When I try to delete just 100 items it works perfectly but when I try a table with 1k records it just errors with the following message:

```
ClientResponseError 0: Something went wrong while processing your request.
    at new ClientResponseError (file:///Users/error-py/Desktop/myProject/node_modules/pocketbase/dist/pocketbase.es.mjs:2:2306)
    at file:///Users/error-py/Desktop/myProject/node_modules/pocketbase/dist/pocketbase.es.mjs:2:45357
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  url: '',
  status: 0,
  response: {},
  isAbort: false,
  originalError: TypeError: fetch failed
      at Object.fetch (node:internal/deps/undici/undici:11457:11)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    cause: Error: connect ECONNRESET 127.0.0.1:8090
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1570:16) {
      errno: -54,
      code: 'ECONNRESET',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 8090
    }
  }
}

```
The table does not have any API rules. I guess its overwhelmed by the amount of traffic..

This is my function that I'm using to delete the records:

```
async function clearTable(tableName: string) {
    await pb.collection(tableName).getFullList({batch: 999999999999999}).then((records) => {
        for (let i = 0; i < records.length; i++) {
            pb.collection(tableName).delete(records[i].id);
        }
        console.log(chalk.green(`Successfully deleted ${chalk.blue(records.length)} records from table: ${tableName}`))
    });
}
```

Ive used this approach because Ive seen a post from some time ago which did it the exact same way

**Comment by 0x1AE7F** [comment]
Ive also tried to use `./pocketbase serve --debug` but it does not show any errors when this happens. After the ECONNRESET it just works fine afterwards and I can send more traffic to pocketbase. It just wont work with that much traffic at once.

**Comment by ganigeorgiev** [comment, marked as answer]
In what environment do you execute the above code? What versions of Node and PocketBase (executable and JS SDK) do you use? Do you have a reverse proxy (eg. nginx)?

I guess this could happen if you've reached the files descriptor limit of the server (you can check ["Increase the open file descriptors limit"](https://pocketbase.io/docs/going-to-production/#increase-the-open-file-descriptors-limit)).

If you are resource constrained, to avoid the error you can try to batch the requests.

**Reply by ganigeorgiev** [reply]
_Side note1: `batch: 999999999999999` will be ignored as the maximum allowed batch size per request is 500._

_Side note2: the console log after the delete will be printed before the delete requests as you are not awaiting the promises. You can add each delete result in an array and then use [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)._

**Reply by 0x1AE7F** [reply]
Oh, okay.. Thanks for the Side notes!
Node Version is: v20.0.0
pocketbase version 0.15.3
I'm using Typescript and this is my script: `"dev": "node --no-warnings --loader ts-node/esm index.ts`
Also I'm running the code and pocketbase on the same M1 MacBook Air 8GB ( arm64 )

**Reply by 0x1AE7F** [reply]
Thank you so much for sending me the "Increase the open file descriptors limit" as this was only set to 256 and setting it to 2048 allowed me to delete around 2k records at once.

**Reply by loganfalzarano** [reply]
I encountered the same issue and I was able to solve it by implementing rate limiting. You can use the pLimit library to put a limit on the number of concurrent operations which helped me precent the connection error.

**Reply by loganfalzarano** [reply]
`
import pLimit from 'p-limit';

//delete all records for the given collection name
const findAndDelete = async (collectionName) => {
  const records = await pb.collection(`${collectionName}`).getFullList({
    sort: '-created',
  });

  const limit = pLimit(5); // Set the maximum number of concurrent operations (e.g., 5)

  await Promise.all(records.map(async (record) => {
    await limit(() => pb.collection(`${collectionName}`).delete(record.id));
  }));
}
`

**Reply by 0x1AE7F** [reply]
Thanks, never really worked with rate limiting ( should have done it a long time ago ) but ill look into the lib that you sent.
